File: python/program/Mahasiswa.py
# Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 2
# dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
# saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

from SivitasAkademik import SivitasAkademik

# class declaration, class Mahasiswa sebagai cucu dari class Human 
# dan anak dari Class Sivitas Akademik
class Mahasiswa(SivitasAkademik):
    
    # Private attributes
    __nim=0
    __prodi=""
    __fakultas=""

    # contructor 
    def __init__(self,nim=0,prodi="",fakultas=""):
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas

    #Getter dan Setter

    # Get nim
    def getNim(self):
        return self.__nim
    
    # set nim
    def setNim(self, nim):
        self.__nim = nim
    
    # get prodi
    def getProdi(self):
        return self.__prodi
    
    # set prodi
    def setProdi(self, prodi):
        self.__prodi = prodi
    
    # get fakultas
    def getFakultas(self):
        return self.__fakultas
    
    # set fakultas
    def setFakultas(self, fakultas):
        self.__fakultas = fakultas
    

File: php/program/Mahasiswa.php
<!-- Saya Amida Zulfa Laila NIM 2101147 mengerjakan Latihan Praktikum 2
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka
saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin. -->
<?php
require ('SivitasAkademik.php');

// Class declaration, class ini sebagai anak dari Sivitas Akademik dan cucu dari Human
class Mahasiswa extends SivitasAkademik {
    // atribut private
    private $nim = '';
    private $prodi = '';
    private $fakultas = '';
    private $foto = '';

    //constructor
    public function __construct( $nim = '', $prodi='', $fakultas='', $foto='' ) {
        $this->nim = $nim;
        $this->prodi = $prodi;
        $this->fakultas = $fakultas;
        $this->foto = $foto;
    }

    // Getter dan Setter
    //set nim
 
    public function setNim($nim){
        $this->nim = $nim;
    }
    //get nim

    public function getNim(){
        return $this->nim;
    }   
    //set prodi
    public function setProdi($prodi){
        $this->prodi = $prodi;
    }
    //get prodi
    public function getProdi(){
        return $this->prodi;
    } 
    //set fakultas
    public function getfakultas(){
        return $this->fakultas;
    }  
    //get fakultas
    public function setFakultas($fakultas){
        $this->fakultas = $fakultas;
    }
    //set foto
    public function getFoto(){
        return $this->foto;
    }
    //get foto
    public function setFoto($foto){
        $this->foto = $foto;
    }
}

?>